### DISCUSSION TITLE
Is there a way to increase Memory overclock higher than +250?

### DISCUSSION THREAD

**Discussion by TheBestPessimist** [seerge/g-helper#654] [Q&A] [answered]
In afterburner, i have a stable overclock at +1550. In Ghelper on the other hand, max is +250.

I have tried to edit the `config.json` file, and set ` "gpu_memory_0": 1000,` and ` "gpu_memory_2": 1000,` (balanced and turbo), but it seems that setting is not applied, nor shown in the GUI.

**Comment by seerge** [comment]
Cause there is a check in the code that would prevent that :) i have deliberately limited ui to those offsets (same as AC does) , as otherwise average user from reddit will crank it up to max and come complaining that its crashing

**Reply by TheBestPessimist** [reply]
roger, so there is no way with Ghelper. thx for confirming. then i'll continue to use afterburner!

**Reply by TheBestPessimist** [reply]
_sidenote: you have a superb response time. heck, i with my APIs would respond that fast <3_

**Reply by seerge** [reply, marked as answer]
I have added 2 optional config params : ``max_gpu_core`` and ``max_gpu_memory`` that you can set. They will affect maximum slider / possible setting values (that default to +250)

[GHelper.zip](https://github.com/seerge/g-helper/files/11801382/GHelper.zip)

**Reply by TheBestPessimist** [reply]
OH. lovely. thank you very much. I will test tomorrow and give feedback.

**Reply by ks76rice** [reply]
How do you edit these paramaters? Not sure where to find the config.json file that was mentioned.

**Reply by TheBestPessimist** [reply]
in appdata ghelper

**Reply by TheBestPessimist** [reply]
@seerge i can finally confirm it works as expected. Thanks!

**Reply by ks76rice** [reply]
I'm still not sure where to find this file. In the G-Helper folder there is only the application, a .config, a .dll, and a .sys file. I see on the main page of Github, there is a NvidiaGpuControl.cs file, but I am not sure what to do with this. Could you walk me through some steps to edit increase memory clock? Thanks!

**Reply by seerge** [reply]
Quit g-helper, open ``%appdata%/ghelper/config.json`` and  add following like to the top (with the comma) ```max_gpu_memory:1200,``` and start app again.

**Reply by ks76rice** [reply]
@seerge 
Thank you! I was able to edit the config file. 
I was playing around with overclocking and noticed something weird. After setting the memory overclock to a value over 300, it would only sometimes "stick" - meaning that the value would show in the G-helper UI, and benchmark tests would show a significant improvement in results.

Other times, these values wouldn't "stick" - meaning that the G-helper UI would show 0 for both overclocks after exiting the UI and coming back to it. Additionally, when I would try to benchmark, the results would be as if there was no overclock applied at all.  I checked the config.json file after this, which showed that the mode that I applied the overclock to DID have the core and memory clocks set to 180, 1000. However, it just would not show in benchmarking results. I then checked the log file, which showed this message after I tried to the set the memory clock to any number over 300:
6/24/2023 1:21:14 AM: SET GPU CLOCKS: 180, 1000
6/24/2023 1:21:14 AM: SET GPU CLOCKS: NVAPI_NOT_SUPPORTED

I am not sure what caused the overclocks to stick sometimes and not stick other times. Also, after setting the memory overclock to +300, everything has functioned fine, even in benchmarks. Since armoury crate has its maximum memory clock slider set to 300, that may have something to do with this? I hope what I wrote makes sense.

**Reply by seerge** [reply]
@ks76rice clock offsets are done via nvidia official api, if it says not supported for a certain values or in certain conditions - probably it is like that :)

**Reply by ks76rice** [reply]
@TheBestPessimist Did you experience anything like this as well?

**Reply by TheBestPessimist** [reply]
I _think_ the values work for me, but i'm also using MSI Afterburner to set the `Power Limit` to 150%, because otherwise my GPU (RTX 4070) does not boost to 100W or higher, instead it stays at around 80-90 W. And yes, there is a 10-20 FPS difference, which makes Cyberpunk go 30-40 => 50-60 FPS

So it's either Afteburner who is doing its job or GHelper.

![image](https://github.com/seerge/g-helper/assets/4482210/4fa34583-8d28-4af8-aae5-7910879b8d54)

**Comment by RahulVivekNair** [comment]
@seerge Are these params available in the master(v0.95.0.0)?, whenever I add max_gpu_memory or max_gpu_core to the config file and run the program, the file gets overwritten and these parameters disappear.

**Reply by seerge** [reply]
@RahulVivekNair you need to edit config before starting the app :)

**Reply by RahulVivekNair** [reply]
hmm strange, I first quit the app , modify the config.json and add these parameters and then save it and run the program again. The added params still get overwritten. Is the code only modified in the file you linked in this thread by any chance?

**Reply by seerge** [reply]
@RahulVivekNair then looks like you don't quit the app prior to editing / saving config.json. App reads config only on start. 

Config params are live in 0.95 (and prior versions too)
https://github.com/seerge/g-helper/blob/main/app/Gpu/NVidia/NvidiaGpuControl.cs#L15

**Reply by RahulVivekNair** [reply]
Thanks,it worked after a reboot!

**Reply by TheBestPessimist** [reply]
@RahulVivekNair i can confirm they work in 0.95
![image](https://github.com/seerge/g-helper/assets/4482210/06b18884-e5dd-4caa-bef0-f83a6bbc0230)

**Reply by MrxSiN** [reply]
So now it confirms its not locked behind vbios. Because last time when i want to OC memory pass 250, i need to swap the vbios. 
This is a nice features. Introduce a toggle to bypass the 250 limit.

**Reply by RahulVivekNair** [reply]
> So now it confirms its not locked behind vbios. Because last time when i want to OC memory pass 250, i need to swap the vbios. 
> This is a nice features. Introduce a toggle to bypass the 250 limit. 

It depends,my 95W 3060 would not oc beyond 300mhz on memory but flashing a 130W vbios did,no locks now.
